kubespawner/objects.py

```diff
@@ -75,63 +103,98 @@ def make_pod_spec(
+    if run_as_uid is not None:
+        security_context.run_as_user = int(run_as_uid)
+    pod.spec.security_context = security_context
+
     if image_pull_secret is not None:
-        image_secret = [{"name": image_pull_secret}]
-    return {
-        'apiVersion': 'v1',
-        'kind': 'Pod',
-        'metadata': {
-            'name': name,
-            'labels': labels,
-        },
-        'spec': {
-            'securityContext': pod_security_context,
-            "imagePullSecrets": image_secret,
-            'containers': [
-                {
-                    'name': 'notebook',
-                    'image': image_spec,
-                    'args': cmd,
-                    'imagePullPolicy': image_pull_policy,
-                    'ports': [{
-                        'containerPort': port,
-                    }],
-                    'resources': {
-                        'requests': {
-                            # If these are None, it's ok. the k8s API
-                            # seems to interpret that as 'no limit',
-                            # which is what we want.
-                            'memory': mem_guarantee,
-                            'cpu': cpu_guarantee,
-                        },
-                        'limits': {
-                            'memory': mem_limit,
-                            'cpu': cpu_limit,
-                        }
-                    },
-                    'env': [
-                        {'name': k, 'value': v}
-                        for k, v in env.items()
-                    ],
-                    'volumeMounts': volume_mounts
-                }
-            ],
-            'volumes': volumes
-        }
-    }
+        pod.spec.image_pull_secrets = []
+        image_secret = V1LocalObjectReference()
+        image_secret.name = image_pull_secret
+        pod.spec.image_pull_secrets.append(image_secret)
+      
+    if node_selector:
+        pod.spec.node_selector = node_selector
+
+    pod.spec.containers = []
+
+    notebook_container = V1Container()
+    notebook_container.name = "notebook"
+    notebook_container.image = image_spec
+    notebook_container.working_dir = working_dir
+    notebook_container.ports = []
+    port_ = V1ContainerPort()
+    port_.name = "notebook-port"
+    port_.container_port = port
+    notebook_container.ports.append(port_)
+    notebook_container.env = []
+    for key, value in env.items():
+        env_ = V1EnvVar()
+        env_.name = key
+        env_.value = value
+        notebook_container.env.append(env_)
+    notebook_container.args = cmd
+    notebook_container.image_pull_policy = image_pull_policy
+    notebook_container.resources = V1ResourceRequirements()
+    notebook_container.resources.requests = {"cpu": cpu_guarantee, "memory": mem_guarantee}
+    notebook_container.resources.limits = {"cpu": cpu_limit, "memory": mem_limit}
+    # notebook_container.liveness_probe = V1Probe()
```

Comment:
Yes, i wanted to have it there temporally in case you wanted it on this PR. It's gone.